node-runner-cli/tests/userflows/install-docker-all-same-host.sh

```diff
@@ -0,0 +1,147 @@
+export COMPOSE_HTTP_TIMEOUT=360
+#Below PATH require when ansible is installed as part of pip
+export PATH="$PATH:/home/ubuntu/.local/bin"
+
+./babylonnode docker config -d $HOME/babylon-node-config \
+  -t ${SEED_NODE} \
+  -m CORE GATEWAY \
+  -n ${NETWORK_ID} \
+  -k ${KEYSTORE_PASSWORD} -nk \
+  -p ${POSTGRES_PASSWORD} \
+  -v "" \
+  -a
+
+cat $HOME/babylon-node-config/config.yaml
+
+# Reset everything
+./babylonnode docker stop -f $HOME/babylon-node-config/config.yaml | true
+./babylonnode monitoring stop | true
+
+# Setup passwords details
+DOCKER_COMPOSE_FOLDER_PREFIX=$(whoami) ./babylonnode auth set-admin-password -m DOCKER -p ${NGINX_ADMIN_PASSWORD}
+DOCKER_COMPOSE_FOLDER_PREFIX=$(whoami) ./babylonnode auth set-metrics-password -m DOCKER -p ${NGINX_METRICS_PASSWORD}
+DOCKER_COMPOSE_FOLDER_PREFIX=$(whoami) ./babylonnode auth set-gateway-password -m DOCKER -p ${NGINX_GATEWAY_PASSWORD}
+
+./babylonnode monitoring config \
+  -m MONITOR_CORE MONITOR_GATEWAY \
+  -cm ${NGINX_METRICS_PASSWORD} \
+  -gm ${NGINX_METRICS_PASSWORD} \
+  -am ${NGINX_METRICS_PASSWORD}
+
+./babylonnode monitoring install -a
+
+sleep 10
+
+grafana_status_code=$(curl -s -o /dev/null -w "%{http_code}\n" localhost:3000/login)
+if [[ $grafana_status_code != 200 ]]; then
+  echo "Grafana did not start properly. Exiting..."
+  exit $grafana_status_code
+fi
+
+prometheus_status_code=$(curl -s -o /dev/null -w "%{http_code}\n" localhost:9090/prometheus/graph)
+if [[ $prometheus_status_code != "200" ]]; then
+  echo "Prometheus did not start properly. Exiting..."
+  exit $prometheus_status_code
+fi
+
+./babylonnode docker install -f $HOME/babylon-node-config/config.yaml -a
+
+echo "Checking container health"
+sleep 10
+docker ps
+
+echo "Waiting and checking again"
+sleep 10
+docker ps
+
+echo "Testing Core node health endpoint"
+set +e
+for i in {1..5}; do
+  FULL_OUTPUT=$(NGINX_ADMIN_PASSWORD=${NGINX_ADMIN_PASSWORD} ./babylonnode api system health)
+  OUTPUT=$(echo $FULL_OUTPUT | jq -r '.status')
+  if [[ $OUTPUT == "SYNCING" || $OUTPUT == "BOOTING_AT_GENESIS" || $OUTPUT == "BOOTING" || $OUTPUT == "UP" ]]; then
+    echo "The result is successful"
+    echo "The Node is in status $OUTPUT"
+    break
+  else
+    if [[ $i == 5 ]]; then
+      echo "failed to get ready in time."
+      echo "here are the logs of the core node"
+      docker logs $(whoami)-core-1 --tail 100
+      echo "Exiting..."
+      exit 137
+    fi
+    echo "The result is unsuccessful. Waiting and trying again ($i of 5)"
+    echo "Command ./babylonnode api system health resulted in"
+    echo "$FULL_OUTPUT"
+  fi
+  sleep 30
+done
+set -e
+
+echo "Testing postgres is set up correctly"
+sudo systemctl status postgresql@12-main.service --no-pager
+PGPASSWORD=$POSTGRES_PASSWORD psql -h localhost -U postgres -d radixdlt_ledger -P pager=off -c "\dt"
+
+echo "Testing gateway endpoints available"
+
+echo "Gateway endpoint"
+curl -k -u "gateway:${NGINX_GATEWAY_PASSWORD}" https://localhost/gateway | jq
```

Comment:
Does Jq fail with exit status that makes job fail?

Reply:
Those commands do not have a cli version yet since they target the gateway. Good ticket for the future actually. Adding the gateway api commands to cli.

Reply:
And yes, the jq will fail if the response is not a valid json. Which is the case if the endpoint is not available, authentication failed or similar.

It wont fail if the gateway returns a valid json that states something like this {status:"not ok"}.
I dont know how that response looks like yet.